### DISCUSSION TITLE
add an explicit option on the activity log to keep it of Cleaning up

### DISCUSSION THREAD

**Discussion by kadevland** [spatie/laravel-activitylog#742] [Ideas]
We need to keep some important log like deletion, creation, or important change of state.

so :

one exemple for manualy
```
activity()->log('Look mum, I logged something')->keepIt()

```
for model event
```
 protected static $keepedRecordEvents = ['deleted'];
```

**Comment by Gummibeer** [comment]
Hey,
for the moment you can use a custom column and the `tap()` or `tapActivity()` method to set your Boolean if the record should be kept.
And create your own cleanup command.

Right now I think that this feature isn't required enough to integrate it into the package and it's relatively easy to get it done.

But I will keep this discussion open if others want it. If so we could check if/when/how we implement it.
